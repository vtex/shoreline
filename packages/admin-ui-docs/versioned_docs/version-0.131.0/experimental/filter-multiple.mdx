---
title: FilterMultiple
path: /filter-multiple/
---

[Filters](/filter/) are supposed to be used in pages or sections that display some kind of data that can be filtered. The `filterMultiple`
 component is a picker that displays filter options and allows for many of them to be selected. When the options are selected
 and the changes applied, it should generate a change on the data or content being displayed, and should be filtered accordingly.

## Usage

```jsx

import {
  experimental_FilterMultiple as FilterMultiple,
  experimental_useFilterMultipleState as useFilterMultipleState,
} from '@vtex/admin-ui'

function Example() {
  const state = useFilterMultipleState({
    items: [
      { label: 'Full', value: 1, id: '#1' },
      { label: 'Empty', value: 2, id: '#2' },
      { label: 'Half empty', value: 4, id: '#3' },
    ],
    onChange: ({ selected }) => console.log(`applied: ${selected}`),
    label: 'Status',
  })
  return <FilterMultiple state={state} />
}
```

## Examples

### Initial values

```jsx noInline

const FilterMultiple = experimental_FilterMultiple
const useFilterMultipleState = experimental_useFilterMultipleState

function Example() {
  const state = useFilterMultipleState({
    items: [
      { label: 'Full', value: 1, id: '#1' },
      { label: 'Empty', value: 2, id: '#2' },
      { label: 'Half full', value: 3, id: '#3' },
      { label: 'Half empty', value: 4, id: '#4' },
      { label: 'Unknown', value: 5, id: '#5' },
    ],
    onChange: ({ selected }) => console.log(`applied: ${selected}`),
    label: 'Status',
    initialApplied: ['#1', '#2'],
  })

  return <FilterMultiple state={state} />


}

render(<Example />)
```

### I18n

The filter-multiple has default translations. Wrap your app with [I18nProvider](/experimental/localization#i18nprovider/) to configure the locale.

```jsx
import {
  experimental_FilterMultiple as FilterMultiple,
  experimental_useFilterMultipleState as useFilterMultipleState,
} from '@vtex/admin-ui'

function Filter() {
  const state = useFilterMultipleState({
    items: [
      { label: 'Full', value: 1, id: '#1' },
      { label: 'Empty', value: 2, id: '#2' },
      { label: 'Half empty', value: 4, id: '#3' },
    ],
    onChange: ({ selected }) => console.log(`applied: ${selected}`),
    label: 'Status',
  })
  return <FilterMultiple state={state} />
}

function RootApp() {
  return (
    <I18nProvider locale="ko-KO">
      <Filter />
    </I18nProvider>
  )
}

render(<RootApp />)
```

## Props

### FilterMultiple

| Name  | Type                  | Description    | Required | Default |
| ----- | --------------------- | -------------- | -------- | ------- |
| state | `FilterMultipleState` | Object that manages the component state logic, you can obtain it calling the useFilterMultipleState hook | âœ…       | -       |

### UseFilterMultipleState

| Name           | Type                             | Description                                                         | Required | Default |
| -------------- | -------------------------------- | ------------------------------------------------------------------- | -------- | ------- |
| items          | `FilterItem[]`                   | List of options for the filter, item must include id and label      | âœ…       | -       |
| onChange       | `({selected: string[]}) => void` | Callback called with current selected ids on apply and clear press. | âœ…       | -       |
| label          | `string`                         | Filter button label                                                 | âœ…       | -       |
| initialApplied | `string[]`                       | Array with initially applied options' ids                           | ðŸš«       | -       |
| baseId         | `string`                         | Optional baseId for controling the components and its childrens id  | ðŸš«       | -       |

## FilterMultipleState (return of UseFilterMultipleState)

| Name           | Type                             | Description
| -------------- | -------------------------------- | -------------------------------------------------------------------
| menu           | `MenuState`                      | State that controls the component's popover menu
| combobox       | `ComboboxState`                  | State that controls the component's option list
| onClear        | `() => void`                     | Function used by the component when clear is selected
| onChange       | `() => void`                     | Function used by the component when apply is selected
| appliedItems   | `FilterItem[]`                   | List of items that are currently applied
| appliedKeys    | `ID[]`                           | List of ids that are currently applied
| selectedKeys   | `ID[]`                           | List of ids that are currently selected
| items          | `FilterItem[]`                   | List of options for the filter


