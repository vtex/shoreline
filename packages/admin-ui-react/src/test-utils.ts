import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
