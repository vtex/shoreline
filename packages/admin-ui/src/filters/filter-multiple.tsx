import React from 'react'
import { tag } from '@vtex/admin-ui-react'
import { Checkbox } from '../checkbox'
import { BaseFilter } from './base-filter'
import { FilterOption } from './filter-option'
import type { UseFilterMultipleReturn } from './base-filter.state'

export function FilterMultiple(props: FilterMultipleProps) {
  const {
    state: { listState, appliedItems = [] },
    state,
  } = props

  const firstSelectedItemLabel =
    appliedItems.length > 1
      ? `${appliedItems[0]?.label},`
      : `${appliedItems[0]?.label}` // no comma in this case

  const remainingSelectedItemsCount =
    appliedItems.length > 1 && `+${appliedItems.length - 1}`

  const appliedValuesLabel = !!appliedItems.length && (
    <>
      <span>:</span>
      <tag.span csx={{ color: '$primary' }}>
        <tag.span
          csx={{
            textOverflow: 'ellipsis',
            overflow: 'hidden',
            whiteSpace: 'nowrap',
            maxWidth: '300px',
            marginX: '$s',
          }}
        >
          {firstSelectedItemLabel}
        </tag.span>
        {remainingSelectedItemsCount}
      </tag.span>
    </>
  )

  const { collection } = listState
  const options = Array.from(collection.getKeys()).map((key) =>
    collection.getItem(key)
  )

  return (
    <BaseFilter state={state} appliedValuesLabel={appliedValuesLabel}>
      {options.map((item) => (
        <FilterOption
          key={item.key}
          item={item}
          state={listState}
          inputRenderer={({ isSelected }) => <Checkbox checked={isSelected} />}
        />
      ))}
    </BaseFilter>
  )
}

export interface FilterMultipleProps {
  state: UseFilterMultipleReturn
}
