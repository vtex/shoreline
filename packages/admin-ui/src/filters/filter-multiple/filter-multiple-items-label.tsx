import React from 'react'
import { tag } from '@vtex/admin-ui-react'
import type { UseFilterMultipleReturn } from '..'
import type { AnyObject } from '../..'

import * as style from '../filter.style'

export const MultipleItemsLabel = (props: {
  appliedItems: AnyObject[]
  state: UseFilterMultipleReturn<any>
}) => {
  const {
    appliedItems,
    state: { getOptionLabel },
  } = props

  const separator = appliedItems.length > 1 ? ',' : ''
  const firstOptionLabel = getOptionLabel(appliedItems[0] || {}) || ''

  const firstSelectedItemLabel = `${firstOptionLabel}${separator}`

  const remainingSelectedItemsCount =
    appliedItems.length > 1 ? `+${appliedItems.length - 1}` : ''

  const appliedValuesLabel = appliedItems.length ? (
    <>
      <span>:</span>
      <tag.span csx={{ color: '$primary' }}>
        <tag.span csx={{ ...style.disclosureStatusLabel, marginX: '$s' }}>
          {firstSelectedItemLabel}
        </tag.span>
        {remainingSelectedItemsCount}
      </tag.span>
    </>
  ) : null

  return appliedValuesLabel
}
