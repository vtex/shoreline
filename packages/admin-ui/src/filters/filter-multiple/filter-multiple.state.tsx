import { useState, useEffect } from 'react'

import type { ComboboxMultipleState } from '../../combobox'
import { useComboboxMultipleState } from '../../combobox'

import type { AnyObject } from '@vtex/admin-ui-util'
import type {
  GenericFilterStateReturn,
  ItemList,
  FilterOption,
} from '../filter/filter.state'
import { useMenuState } from 'ariakit'
import { useFilterStatus } from '../use-filter-status'

export function useFilterMultipleState<T extends AnyObject>(
  props?: UseFilterMultipleStateProps<T>
): UseFilterMultipleReturn<T> {
  const { searchableList: initialSearchableList = [] } = props || {}

  const [appliedItems, setAppliedItems] = useState<ItemList<T>>([])
  const [searchableList, setSearchableList] = useState(initialSearchableList)

  const comboboxMultiple = useComboboxMultipleState<FilterOption<T>>({
    list: searchableList,
    getOptionValue: (option) => option.label,
    compare: (optionA, optionB) => optionA.id === optionB.id,
  })

  const menu = useMenuState({ ...comboboxMultiple, gutter: 8 })
  const { status, setStatus } = useFilterStatus(comboboxMultiple)

  useEffect(() => {
    comboboxMultiple.setSelectedItems([])
  }, [])

  const { selectedItems } = comboboxMultiple

  const updateApplied = (items: ItemList<T>) => {
    comboboxMultiple.setSelectedItems(items)
    setAppliedItems(items)
  }

  const apply = () => {
    setAppliedItems(selectedItems)
    menu.hide()
  }

  const clear = () => {
    comboboxMultiple.clearSelected()

    setAppliedItems([])
    menu.hide()
  }

  useEffect(() => {
    const isMenuClosed = !menu.mounted
    const hasSelectedItem = comboboxMultiple.value || selectedItems?.length

    if (isMenuClosed && hasSelectedItem) {
      // resets combobox
      comboboxMultiple.setValue('')
      comboboxMultiple.setSelectedItems(appliedItems)
    }
  }, [menu.mounted])

  useEffect(() => {
    comboboxMultiple.setMatches(searchableList)
  }, [searchableList])

  const getFromApplied = (key: string) => {
    return appliedItems
      .map((item) => item?.[key as keyof FilterOption<T>] || item?.value?.[key])
      .filter((item) => item !== undefined)
  }

  const {
    matches,
    setMatches,
    deferredValue: deferredSearchValue,
    value: searchValue,
  } = comboboxMultiple

  return {
    combobox: comboboxMultiple,
    onClear: clear,
    onChange: apply,
    appliedItems,
    setAppliedItems: updateApplied,
    selectedItems,
    menu,
    getFromApplied,
    status,
    setStatus,
    setSearchableList,
    matches,
    setMatches,
    deferredSearchValue,
    searchValue,
  }
}

export interface UseFilterMultipleReturn<T>
  extends GenericFilterStateReturn<T> {
  combobox: ComboboxMultipleState<FilterOption<T>>
  appliedItems: ItemList<T>
  selectedItems: ItemList<T>
  setAppliedItems: (items: ItemList<T>) => void
  getFromApplied: (key: string) => any[]
}

export interface UseFilterMultipleStateProps<T> {
  /** List of items to be showed on the list. */
  searchableList?: ItemList<T>
}
