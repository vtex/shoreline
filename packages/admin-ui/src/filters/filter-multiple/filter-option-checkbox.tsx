import React from 'react'

import { ComboboxItem } from 'ariakit/combobox'
import { Checkbox } from '../../checkbox'

import * as style from '../filter.style'
import { createComponent, useElement } from '@vtex/admin-ui-react'
import { usePopoverContext } from '../filter-popover-context'
import type { ComboboxMultipleState } from '../../combobox'
import type { FilterOption } from '../filter/filter.state'

export const FilterOptionCheckbox = createComponent<
  typeof ComboboxItem,
  FilterOptionCheckboxProps
>((props) => {
  const { id, value, label, ...restProps } = props
  const { state } = usePopoverContext()

  const item = { id, label, value }

  const combobox = state.combobox as ComboboxMultipleState<FilterOption<any>>

  return useElement(ComboboxItem, {
    baseStyle: style.option,
    children: (
      <>
        <Checkbox
          checked={combobox.isSelected(item)}
          aria-checked={undefined}
          csx={{ marginRight: '$m' }}
          readOnly
        />
        {label}
      </>
    ),
    'aria-selected': combobox.isSelected(item),
    onClick: () => combobox.onChange(item),
    id,
    ...restProps,
  })
})

interface FilterOptionCheckboxProps {
  id: string
  label: string
  value?: any
}
