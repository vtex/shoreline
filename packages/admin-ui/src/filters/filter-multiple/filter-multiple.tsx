import React from 'react'
import { Checkbox } from '../../checkbox'
import { BaseFilter } from '../filter-base'
import { ComboboxItem } from '../../combobox/combobox-item'
import { MultipleItemsLabel } from './filter-multiple-items-label'

import type { UseFilterMultipleReturn } from './filter-multiple.state'

import * as style from '../filter.style'

export function FilterMultiple<T>(props: FilterMultipleProps<T>) {
  const {
    state: {
      appliedItems,
      items,
      combobox,
      baseId,
      getOptionId,
      getOptionLabel,
    },
    state,
  } = props

  return (
    <BaseFilter
      state={state}
      appliedValuesLabel={
        <MultipleItemsLabel appliedItems={appliedItems} state={state} />
      }
    >
      {items.map((item) => (
        <ComboboxItem
          aria-selected={combobox.isSelected(item)}
          key={getOptionId(item)}
          onClick={() => combobox.onChange(item)}
          style={style.option}
          id={`${baseId ?? ''}-item-${getOptionId(item)}`}
        >
          <Checkbox
            id={getOptionId(item)}
            checked={combobox.isSelected(item)}
            aria-checked={undefined}
            csx={{ marginRight: '$s' }}
            label={getOptionLabel(item)}
            aria-readonly="true"
          />
        </ComboboxItem>
      ))}
    </BaseFilter>
  )
}

export interface FilterMultipleProps<T> {
  state: UseFilterMultipleReturn<T>
}
