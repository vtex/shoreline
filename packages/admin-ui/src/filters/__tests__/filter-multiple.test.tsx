import React from 'react'
import { useFilterMultipleState } from '..'
import { render, axe, withState } from '../../test-utils'

import { FilterMultiple } from '../filter-multiple/filter-multiple'

const FilterWithState = withState(FilterMultiple, () =>
  useFilterMultipleState({
    items: [
      { label: 'Full', value: 1, id: '#1' },
      { label: 'Empty', value: 2, id: '#2' },
    ],
    label: 'Filter',
    baseId: 'filterMultipleTest',
  })
)

describe('Multiselect filter tests', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(<FilterWithState />)

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
