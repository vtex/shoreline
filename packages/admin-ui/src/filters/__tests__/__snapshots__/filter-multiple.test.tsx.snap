// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiselect filter tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: #2953B2;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
  margin-right: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  background: #F4F4F4;
  color: #545454;
}

.emotion-3:hover {
  color: #FFFFFF;
  background: #20408C;
}

.emotion-3:active {
  color: #FFFFFF;
  background: #172F66;
}

.emotion-3:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:hover {
  background: #E6E6E6;
  color: #545454;
}

.emotion-3:active {
  background: #DDDDDD;
  color: #545454;
}

.emotion-3 svg {
  margin-left: 0.125rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 svg {
  margin-right: 0.25rem;
}

.emotion-0 {
  height: 1rem;
  width: 1rem;
  min-height: 1rem;
  min-width: 1rem;
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  margin-left: 0.25rem;
}

<button
    aria-expanded="false"
    aria-haspopup="menu"
    class="emotion-3"
    data-command=""
    data-disclosure=""
    id="filterMultipleTest-disclosure"
    type="button"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-1"
      >
        Filter
        <svg
          class="emotion-0"
          fill="currentColor"
          height="1rem"
          viewBox="0 0 256 256"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="16rem"
            width="16rem"
          />
          <polyline
            fill="none"
            points="48 160 128 80 208 160"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="18"
          />
        </svg>
      </div>
    </div>
  </button>
  .emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-14 svg {
  margin-right: 0.25rem;
}

.emotion-21 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  box-shadow: 0rem 0rem 1.5rem 0rem rgba(0,0,0,0.05);
  border: 1px solid #DDDDDD;
  border-radius: 4px;
  background: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 256px;
  z-index: 999;
}

.emotion-21:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-21:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-13 {
  margin-top: 0.25rem;
  padding: 1rem 1.25rem;
  max-height: 312px;
  overflow-y: auto;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-12 >*:not(:first-child) {
  margin-top: 1rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-3 >* {
  margin-left: 0.5rem;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 1.25rem;
  height: 1.25rem;
  background: form.control;
  border: 1px solid #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  cursor: pointer;
  margin: 0rem;
  box-sizing: border-box;
  margin-right: 0.25rem;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:hover:not(:disabled) {
  border: 1px solid #545454;
}

.emotion-0:focus-visible {
  outline: none;
  border: 1px solid #545454;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked {
  background-color: #2953B2;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-0:checked:focus-visible {
  outline: none;
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-0:checked:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23FFFFFF"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-0:checked[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23747474"/></svg>');
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:indeterminate:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23000000"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-0:indeterminate[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23747474"/></svg>');
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-2 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-20 {
  border-top: none;
  padding: 1rem 1.25rem;
  padding-top: 0rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
}

.emotion-20 button:not(:first-child) {
  margin-left: 0.75rem;
}

.emotion-16 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #2953B2;
  background: transparent;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
}

.emotion-16:hover {
  color: #20408C;
  background: #F1F4FC;
}

.emotion-16:active {
  color: #172F66;
  background: #DDE5F7;
}

.emotion-16:disabled {
  background: transparent;
  color: #747474;
}

.emotion-16:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-16:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-19 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 500;
  font-size: 0.875rem;
  line-height: 1.1428571429;
  letter-spacing: 0.01em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: #2953B2;
  padding: 0.5rem 0.75rem;
  height: 2.25rem;
}

.emotion-19:hover {
  color: #FFFFFF;
  background: #20408C;
}

.emotion-19:active {
  color: #FFFFFF;
  background: #172F66;
}

.emotion-19:disabled {
  background: #E6E6E6;
  color: #747474;
}

.emotion-19:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-19:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

<div
    role="presentation"
    style="position: absolute; left: 0px; top: 0px; margin: 0px;"
  >
    <div
      aria-orientation="vertical"
      class="emotion-21"
      data-dialog=""
      hidden=""
      id="filterMultipleTest-popover"
      role="menu"
      style="position: relative; display: none;"
      tabindex="0"
    >
      <span
        class="emotion-13"
      >
        <div
          class="emotion-12"
          hidden=""
          id="filterMultipleTest-combobox-list"
          role="listbox"
          style="display: none;"
        >
          <div
            aria-selected="false"
            class="emotion-5"
            data-command=""
            data-composite-hover=""
            id="filterMultipleTest-item-#1"
            style="display: flex; cursor: pointer;"
            tabindex="0"
          >
            <div
              class="emotion-4"
            >
              <div
                class="emotion-3"
              >
                <input
                  aria-readonly="true"
                  class="emotion-0"
                  data-command=""
                  id="#1"
                  type="checkbox"
                />
                <div
                  class="emotion-2"
                >
                  <label
                    class="emotion-1"
                    for="#1"
                  >
                    Full
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-selected="false"
            class="emotion-5"
            data-command=""
            data-composite-hover=""
            id="filterMultipleTest-item-#2"
            style="display: flex; cursor: pointer;"
            tabindex="0"
          >
            <div
              class="emotion-4"
            >
              <div
                class="emotion-3"
              >
                <input
                  aria-readonly="true"
                  class="emotion-0"
                  data-command=""
                  id="#2"
                  type="checkbox"
                />
                <div
                  class="emotion-2"
                >
                  <label
                    class="emotion-1"
                    for="#2"
                  >
                    Empty
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </span>
      <div
        class="emotion-20"
      >
        <button
          class="emotion-16"
          type="button"
        >
          <div
            class="emotion-15"
          >
            <div
              class="emotion-14"
            >
              Clear
            </div>
          </div>
        </button>
        <button
          class="emotion-19"
          type="button"
        >
          <div
            class="emotion-15"
          >
            <div
              class="emotion-14"
            >
              Apply
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
